VMware.CloudFoundation.PowerManagement.psm1

```diff
@@ -744,6 +744,101 @@ Function Set-VsanClusterPowerStatus {
+        The cmdlet will perform a dry run test prior to initiate a shutdown command on a VxRail cluster.
+
+        .EXAMPLE
+        Invoke-VxrailClusterShutdown -server sfo-w01-vxrm.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1!
+        This example powers off a VxRail cluster cluster which is managed by the VxRail Manager sfo-w01-vxrm.sfo.rainpole.io controls. 
+
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass
+    )
+	
+    Try {
+        Write-PowerManagementLogMessage -Type INFO -Message "Starting the call to the Invoke-VxrailClusterShutdown cmdlet." -Colour Green
+
+        $checkServer = (Test-NetConnection -ComputerName $server -Port 443).TcpTestSucceeded
+        if ($checkServer) {
+            Write-PowerManagementLogMessage -Type INFO -Message "Connecting to '$server'..."
+            if ($DefaultVIServers) {
+                Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue -ErrorAction  SilentlyContinue | Out-Null
+            }
+
+            # Prepare VxRail rest API headers and payload
+            $payloadTest = @{ dryrun = 'true'} | ConvertTo-Json
+            $payloadRun = @{ dryrun = 'false'} | ConvertTo-Json
+            $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $user, $pass))) # Create Basic Authentication Encoded Credentials
+            $headers = @{"Content-Type" = "application/json" }
+            $headers.Add("Authorization", "Basic $base64AuthInfo")
+			$uri = "https://$server/rest/vxm/v1/cluster/shutdown"
+
+            Write-PowerManagementLogMessage -Type INFO -Message "Starting VxRail cluster shutdown dry run." -Colour Green
+			$respond = Invoke-WebRequest -Method POST -Uri $uri -Headers $headers -Body $payloadTest -UseBasicParsing
+            if($respond.StatusCode -eq "202" -or $respond.StatusCode -eq "200") {
+                $requestID = $respond.content | ConvertFrom-JSON
+                Write-PowerManagementLogMessage -Type INFO -Message "VxRail cluster shutdown request accepted(ID:$($requestID.request_id))"
+                $uri2 = "https://$server/rest/vxm/v1/requests/$($requestID.request_id)"
+                $loopCounter = 0
+                while ($loopCounter -lt 13){
+                    $respond2 = Invoke-WebRequest -Method GET -Uri $uri2 -Headers $headers -UseBasicParsing
+                    if ($respond2.StatusCode -eq "202" -or $respond2.StatusCode -eq "200") {
+                        $checkProgress = $respond2.content | ConvertFrom-JSON
+                        if($checkProgress.state -Match "COMPLETED" -or $checkProgress.state -Match "FAILED" ) {
+                            break
+                        }
+                    }
+                    sleep(10)
```

Comment:
Inconsistent in use of sleep and the naming conventions for the counter.

Reply:
addressed in commit a0e6a231b99ac37596cd1b07cf674e22d71bda11